Sources/Workspace/Workspace.swift

```diff
@@ -731,9 +733,20 @@ extension Workspace {
         packages: [AbsolutePath],
         diagnostics: DiagnosticsEngine
     ) -> [Manifest] {
-        let rootManifests = packages.compactMap({ package -> Manifest? in
-            loadManifest(packagePath: package, url: package.pathString, packageKind: .root, diagnostics: diagnostics)
-        })
+
+        let lock = Lock()
```

Comment:
👀 this is a small but important performance optimization

Reply:
In case there did turn out to be issues with concurrently loading package manifests, this looks as if it could fairly easily be conditionalized back to being serial.  I don't think we need / should do that now, but that possibility makes this safer.

Question:  is there any limit to this concurrency?  I don't think the queue supports that so if I had, say, 100 manifests to evaluate, they might all get compiled and run at once?

Reply:
yup, we could also use an operation queue with limit on concurrency. for the record tested with fairly large projects (~50 dependencies)